Mage/Task/BuiltIn/Deployment/Strategy/GitRebaseTask.php

```diff
@@ -53,8 +53,10 @@ public function run()
         $result = $this->runCommandRemote($command);
 
         if ($result === false) {
-            $repository = $this->getConfig()->deployment('repository');
-            if ($repository) {
+            $scmConfig = $this->getConfig()->deployment('scm');
```

Comment:
Well I took a look on it and it's a little mess here. As you said, it's not well documented, example configs are inconsistent and things are getting weird when you try to run `GitRebase` strategy deployment and `Scm\Clone` task. So I think we need discuss more about it with author. Probably we need some BC change to refactor it. 
But for now, in my opinion, we should mix parameters of deployment task and normal one because let's say you deploy your application from source X but you also want to clone Y into another place to execute some script or update some workers running by shared message queueing application (RabbitMq, for example). So I would let it be as it is now in production code and wait for some decisions about project's future. What do you think?